src/Marten/Schema/JsonLocatorField.cs

```diff
@@ -14,38 +15,55 @@ public static JsonLocatorField For<T>(EnumStorage enumStyle, Casing casing, Expr
         {
             var property = ReflectionHelper.GetProperty(expression);
 
-            return new JsonLocatorField("d.data", new StoreOptions(), enumStyle, casing, property);
+            return new JsonLocatorField("d.data", null, enumStyle, casing, property);
         }
 
         private readonly Func<Expression, object> _parseObject = expression => expression.Value();
 
-        public JsonLocatorField(string dataLocator, StoreOptions options, EnumStorage enumStyle, Casing casing, MemberInfo member) : base(enumStyle, member)
+        public JsonLocatorField(string dataLocator, string databaseSchemaName, EnumStorage enumStyle, Casing casing, MemberInfo member, string pgType = null) :
+            this(dataLocator, databaseSchemaName, enumStyle, casing, new[] { member }, pgType)
         {
-            var memberType = member.GetMemberType();
-            var memberName = member.Name.FormatCase(casing);
+        }
+
+        public JsonLocatorField(string dataLocator, string databaseSchemaName, EnumStorage enumStyle, Casing casing, MemberInfo[] members, string pgType = null) : base(enumStyle, members)
+        {
+            var locator = new StringBuilder(dataLocator);
+            var depth = 1;
+            foreach (var memberInfo in members)
+            {
+                locator.Append(depth == members.Length ? " ->> " : " -> ");
+                locator.Append($"'{memberInfo.Name.FormatCase(casing)}'");
+                depth++;
+            }
+
+            var isStringEnum = MemberType.IsEnum && enumStyle == EnumStorage.AsString;
 
-            var isStringEnum = memberType.IsEnum && enumStyle == EnumStorage.AsString;
-            if (memberType == typeof(string) || isStringEnum)
+            if (!string.IsNullOrWhiteSpace(pgType))
             {
-                SqlLocator = $"{dataLocator} ->> '{memberName}'";
+                PgType = pgType;
             }
-            else if (TypeMappings.TimespanTypes.Contains(memberType))
+
+            if (MemberType == typeof(string) || isStringEnum)
+            {
+                SqlLocator = $"{locator}";
+            }
+            else if (TypeMappings.TimespanTypes.Contains(MemberType))
             {
-                SqlLocator = $"{options.DatabaseSchemaName}.mt_immutable_timestamp({dataLocator} ->> '{memberName}')";
-                SelectionLocator = $"CAST({dataLocator} ->> '{memberName}' as {PgType})";
+                SqlLocator = $"{databaseSchemaName ?? StoreOptions.DefaultDatabaseSchemaName}.mt_immutable_timestamp({locator})";
```

Comment:
**FIX:** If the `options` parameter from the constructor was removed then `StoreOptions` won't be initialized and available.

Reply:
`options` restored, but this was just using a static const for the default value.